ewatercycle/models/pcrglobwb.py

```diff
@@ -15,94 +17,175 @@
+        if value is not None and Path(value).exists():
+            self.__dict__[name] = Path(value).expanduser().resolve()
+        else:
+            warnings.warn(
+                f"No valid input path received for {name}, defaulting to ''.")
+            self.__dict__[name] = ''
+
+
 class PCRGlobWB(AbstractModel):
     """eWaterCycle implementation of PCRGlobWB hydrological model.
 
-    Attributes
+    Args:
+
+        version: pick a version from :py:attr:`~available_versions`
+        parameter_set: instance of :py:class:`~PCRGlobWBParameterSet`.
+
+    Attributes:
+
         bmi (Bmi): GRPC4BMI Basic Modeling Interface object
     """
-    def setup(  # type: ignore
-            self,
-            input_dir: PathLike,
-            cfg_file: PathLike,
-            additional_input_dirs: Iterable[PathLike] = [],
-            **kwargs) -> Tuple[PathLike, PathLike]:
-        """Start model inside container and return config file and work dir.
+    available_versions = ('setters')
 
-        Args:
+    def __init__(self,
+                 version: str,
+                 parameter_set: PCRGlobWBParameterSet,
+                 forcing: Optional[PCRGlobWBForcing] = None):
+        super().__init__()
 
-            - input_dir: main input directory. Relative paths in the cfg_file
-            should start from this directory.
+        self.version = version
+        self.parameter_set = parameter_set
+        self._additional_input_dirs: Iterable[PathLike] = []
 
-            - cfg_file: path to a valid pcrglobwb configuration file,
-            typically somethig like `setup.ini`.
+        self._set_docker_image()
+        self._set_singularity_image()
 
-            - additional_input_dirs: one or more additional data directories
-            that the model will have access to.
+        self._setup_work_dir()
+        self._setup_default_config()
 
-            - **kwargs (optional, dict): any settings in the cfg_file that you
-            want to overwrite programmatically. Should be passed as a dict,
-            e.g. `meteoOptions = {"temperatureNC": "era5_tas_1990_2000.nc"}`
-            where meteoOptions is the section in which the temperatureNC option
-            may be found.
+        if forcing is not None:
+            self._update_config(meteoOptions=forcing.__dict__)
 
-        Returns: Path to config file and work dir
-        """
-        self._setup_work_dir()
-        self._setup_config(cfg_file, input_dir, **kwargs)
-        self._start_container(input_dir, additional_input_dirs)
+    def _set_docker_image(self):
+        images = {
+            "setters": "ewatercycle/pcrg-grpc4bmi:setters",
+        }
+        self.docker_image = images[self.version]
 
-        return self.cfg_file, self.work_dir,
+    def _set_singularity_image(self):
+        # TODO auto detect sif file based on docker image and singularity dir.
+        images = {
+            "setters": "ewatercycle/pcrg-grpc4bmi:setters",
```

Comment:
This does not look like a correct image name for singularity. 
Can it be changed to a sif file or `docker://ewatercycle/pcrg-grpc4bmi:setters`?